### ISSUE TITLE
Out of range dropdown options should be disabled

### ISSUE THREAD

**Issue by ZakisM** [gpbl/react-day-picker#1814]
### Bug description

The dropdowns that are out of range should be disabled if they are after toDate.

### To reproduce

https://codesandbox.io/s/react-daypicker-forked-ymyrvh

#### Steps

1. On the first calendar, set the month to December and the Year to 2025.

### Expected behavior

The dropdown for the month January and dropdown for the year 2026 should be disabled, as this is out of range. The arrow is disabled correctly in this case, but the dropdowns are not.

### Screenshots

<img width="613" alt="Screenshot 2023-06-20 at 11 18 38" src="https://github.com/gpbl/react-day-picker/assets/8143258/9fe9195c-ea73-4ddd-b5a6-7aa59b673597">

### Additional context

It's possible to fix this currently but it requires duplicating a lof the components and creating a custom `Caption` component. If we had access to the `displayMonth` prop inside of the `Dropdown` component it would make it possible to disable this, which would also be useful:

For example a custom Dropdown Component with our own `data-disabled` prop.

```tsx
const DropdownCustom = (props: DropdownCustomPropsType): ReactElement => {
    const { onChange, value, children, caption, className, style } = props;
    const dayPicker = useDayPicker();

    const IconDropdownComponent = dayPicker.components?.IconDropdown ?? IconDropdown;

    return (
        <div className={className} style={style} data-disabled={isAfter(displayMonth, dayPicker.toDate!)}>
            <span className={dayPicker.classNames.vhidden}>{props["aria-label"]}</span>
            <select
                name={props.name}
                aria-label={props["aria-label"]}
                className={dayPicker.classNames.dropdown}
                style={dayPicker.styles.dropdown}
                value={value}
                onChange={onChange}
                disabled={disabled === true ? true : undefined}>
                {children}
            </select>
            <div
                className={dayPicker.classNames.caption_label}
                style={dayPicker.styles.caption_label}
                aria-hidden="true">
                {caption}
                {
                    <IconDropdownComponent
                        className={dayPicker.classNames.dropdown_icon}
                        style={dayPicker.styles.dropdown_icon}
                    />
                }
            </div>
        </div>
    );
};`

**Comment by gpbl** [comment]
Hey @ZakisM thanks for your report, definitely there's space for improvements here. Will check it out better later. 

> If we had access to the displayMonth prop inside of the Dropdown component it would make it possible to disable this, which would also be useful:

There's a way to know the `displayMonths`: the [`useNavigation` hook](https://react-day-picker.js.org/api/interfaces/NavigationContextValue#displaymonths) returns that information and it is accessible by a custom component:

https://codesandbox.io/s/react-daypicker-forked-62vjs7?file=/src/App.tsx:213-314

```tsx
import { useNavigation } from "react-day-picker";

function CustomDropdown(props: DropdownProps) {
  const navigation = useNavigation();
  console.log("Displayed months: ", navigation.displayMonths);
  return <Dropdown {...props} />;
}
```

**Comment by ZakisM** [comment]
Hey, I think I tried this, however when you have multiple months an array is returned. How would I then know which index I need to use for the CustomDropdown?

**Comment by ZakisM** [comment]
Ah I see, the issue is that when the dropdowns are not disabled, if I pick one of the months from the dropdown, the date picker is now showing dates that are outside of the fromDate/toDate I set:


https://github.com/gpbl/react-day-picker/assets/8143258/189d2e0a-b3ca-4c1a-879f-21f82a1699a9

**Comment by gpbl** [comment]
With https://github.com/gpbl/react-day-picker/issues/1853 I'm adding the "displayIndex" to the Caption components. This could help fixing this issue.

**Comment by desire** [comment]
@gpbl bumping this as i really need this fixed

**Comment by dmvdven** [comment]
> Ah I see, the issue is that when the dropdowns are not disabled, if I pick one of the months from the dropdown, the date picker is now showing dates that are outside of the fromDate/toDate I set:
> 
>  recording.mov

I do not use the "disabled" prop as @ZakisM does, so when I just use the fromDate and toDate prop and I select a month which is out of range, the calendar does not show at all. Therefor I think that months that are out of range, should be disabled in the dropdown.

**Comment by gpbl** [comment]
Good news: this is being fixed with #2194 (next beta release)

<img width="810" alt="Screenshot 2024-06-05 at 7 27 10 PM" src="https://github.com/gpbl/react-day-picker/assets/120693/9f92d875-36d1-49b4-9afb-0f4fa0a7a4ee">
